bootstrap_cfn/elb.py

```diff
@@ -0,0 +1,77 @@
+import sys
+import logging
+import boto.ec2.elb
+from bootstrap_cfn import cloudformation
+from bootstrap_cfn import iam
+from bootstrap_cfn import utils
+from bootstrap_cfn.errors import CloudResourceNotFoundError
+
+
+class ELB:
+
+    cfn = None
+    iam = None
+    aws_region_name = None
+    aws_profile_name = None
+
+    def __init__(self, aws_profile_name, aws_region_name='eu-west-1'):
+        self.aws_profile_name = aws_profile_name
+        self.aws_region_name = aws_region_name
+
+        self.conn_elb = utils.connect_to_aws(boto.ec2.elb, self)
+
+        self.iam = iam.IAM(aws_profile_name, aws_region_name)
+        self.cfn = cloudformation.Cloudformation(
+            aws_profile_name, aws_region_name
+        )
+
+    def set_ssl_certificates(self, ssl_config, stack_name):
+        """
+        Look for SSL listeners on all the load balancers connected to
+        this stack, then set update the certificate to that of the config
+
+        Args:
+            ssl_config (dictionary): Certification names to corresponding data
+            stack_name (string): Name of the stack
+
+        Raises:
+            CloudResourceNotFoundError: Raised when the load balancer key in the cloud
+                config is not found
+        """
+        for cert_name in ssl_config.keys():
+            # Get the cert id and also its arn
+            cert_id = "{0}-{1}".format(cert_name, stack_name)
+            cert_arn = self.iam.get_arn_for_cert(cert_id)
+
+            # Get all stack load balancers
+            load_balancer_resources = self.cfn.get_stack_load_balancers(stack_name)
+            found_load_balancer_names = [lb.physical_resource_id for lb in load_balancer_resources]
+            # Use load balancer names to filter gettingload balancer details
+            if len(found_load_balancer_names) > 0:
```

Comment:
think we might want `load_balancers=[]` here or in the else block because if this is not true the next if statement will fail (if I'm reading it right).